### PR TITLE
Fixing documentation for BIP 340-related functions

### PR DESCRIPTION
**Pull request by dr-orlovsky**

### CODE REVIEW

**Comment by apoelstra** [comment]
nit: in one place you wrote `si` instead of `is`.

needs rebase.

concept ACK

**Comment by dr-orlovsky** [comment]
@apoelstra rebased & fixed

File: src/schnorrsig.rs [hunk 1]
```diff
@@ -142,9 +141,13 @@ impl KeyPair {
...
     /// Creates a Schnorr KeyPair directly from generic Secp256k1 secret key
     ///
-    /// Panics if internal representation of the provided [`SecretKey`] does not
-    /// holds correct secret key value obtained from Secp256k1 library
-    /// previously
+    /// # Panic
+    ///
+    /// Panics if internal representation of the provided [`SecretKey`] does not hold correct secret
+    /// key value obtained from Secp256k1 library previously, for instance when the corresponding
+    /// public key is not even key.
```
Comment (apoelstra) [inline src/schnorrsig.rs:148]: To be clearer: this will panic on out-of-range keys (0 or in excess of the group order).

File: src/schnorrsig.rs [hunk 2]
```diff
@@ -142,9 +141,13 @@ impl KeyPair {
...
+    // TODO: Consider returning `Result`, since even if the secret key is constructed with Secp256k1
+    //       library, the function still will panic on keys producing odd public key values.
```
Comment (apoelstra) [inline src/schnorrsig.rs:150]: This doesn't sound right. AFAIK if the resultant public key were to be odd, it will simply be negated.

Comment (apoelstra) [reply src/schnorrsig.rs:150]: No, it's even simpler than that, the `keypair` type simply has no restrictions on key parity.

**Comment by dr-orlovsky** [comment]
@apoelstra addressed your comments and rebased

**Review by apoelstra — APPROVED** [review]
ACK 931b560dc7a339c7544a1eb719b1e4d450456836

**Pull request merged**